x-pack/plugins/apm/common/utils/queries.ts

```diff
@@ -5,19 +5,41 @@
  * 2.0.
  */
 
-import { ESFilter } from '../../../../../../typings/elasticsearch';
+import { ESFilter } from '../../../../typings/elasticsearch';
 import {
   ENVIRONMENT_NOT_DEFINED,
   ENVIRONMENT_ALL,
-} from '../../../../common/environment_filter_values';
-import { SERVICE_ENVIRONMENT } from '../../../../common/elasticsearch_fieldnames';
+} from '../environment_filter_values';
+import { SERVICE_ENVIRONMENT } from '../elasticsearch_fieldnames';
 
-export function getEnvironmentUiFilterES(environment?: string): ESFilter[] {
+type QueryContainer = ESFilter;
+
+export function environmentQuery(environment?: string): QueryContainer[] {
   if (!environment || environment === ENVIRONMENT_ALL.value) {
     return [];
   }
+
   if (environment === ENVIRONMENT_NOT_DEFINED.value) {
     return [{ bool: { must_not: { exists: { field: SERVICE_ENVIRONMENT } } } }];
   }
+
   return [{ term: { [SERVICE_ENVIRONMENT]: environment } }];
 }
+
+export function rangeQuery(
+  start: number,
+  end: number,
+  field = '@timestamp'
+): QueryContainer[] {
+  return [
+    {
+      range: {
+        [field]: {
+          gte: start,
+          lte: end,
+          format: 'epoch_millis',
+        },
+      },
+    },
+  ];
+}
```

Comment:
what's the reasoning for returning an array instead if the object?

Reply:
Because `filter: [{}]` is an error in Elasticsearch and `filter: [...[]]` is not. I would prefer to return an object but because we're typically spreading these (as was the case with what is now `environmentFilter`) returning an array works better.